SalesforceFoundation/HEDAP
# Acc naming enhancement

# Critical Changes

# Changes
- We are adding enhancements for account naming
  - Avoid null value comparison for custom setting: Automatic_Household_Naming__c
  - Check if record type id is null
  - When deleting a contact, automatic naming only applies on household account when it is enabled
# Issues Closed
Fixes #621
# New Metadata

# Deleted Metadata

# Testing Notes
Issue 1:
1. Go to HEDA setting page and set "Administrative" as default account model
2. Create a test contact: test contact 1
3. Create another test contact under the parent account of test contact 1 
4. Delete test contact 1 and make sure the name of parent account is not changed

Issue 2: 
1. Go to HEDA setting page and set "Administrative" as default account model
2. Create a test contact: test contact 1
3. Under record page of test contact 1, clear up account field
4. Make sure the parent account of test contact 1 is not renamed

Issue 3:
1. Create a different record type under account: test record type
2. Go to HEDA setting page and set "Household" as "Default account model"
3. Set "test record type" as the "Household Account Record Type"
4. Make sure Household account naming is still working

Issue 4: 
1. Go to Setup -> Custom Settings -> Manage Hierarchy settings
2. Clear up Automatic_Household_Naming__c
3. Household account should be deactivated right now but make sure no null pointer exception is thrown
